pkg/services/ngalert/api.go

```diff
@@ -0,0 +1,183 @@
+package ngalert
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/eval"
+
+	"github.com/go-macaron/binding"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/api"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/middleware"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/tsdb"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+func (ng *AlertNG) registerAPIEndpoints() {
+	ng.RouteRegister.Group("/api/alert-definitions", func(alertDefinitions routing.RouteRegister) {
+		alertDefinitions.Get("", middleware.ReqSignedIn, api.Wrap(ng.listAlertDefinitions))
+		alertDefinitions.Get("/eval/:alertDefinitionId", ng.validateOrgAlertDefinition, api.Wrap(ng.alertDefinitionEval))
+		alertDefinitions.Post("/eval", middleware.ReqSignedIn, binding.Bind(evalAlertConditionCommand{}), api.Wrap(ng.conditionEval))
+		alertDefinitions.Get("/:alertDefinitionId", ng.validateOrgAlertDefinition, api.Wrap(ng.getAlertDefinitionEndpoint))
+		alertDefinitions.Delete("/:alertDefinitionId", ng.validateOrgAlertDefinition, api.Wrap(ng.deleteAlertDefinitionEndpoint))
```

Comment:
good point; I think it's redundant because if not signed-in then `c.OrgId` is zero so this condition becomes false and `403 Forbidden` is returned:
https://github.com/grafana/grafana/blob/b59045aa21f1aa68e829d62a48814b0348de8a03/pkg/services/ngalert/middleware.go#L16